lib/diagnostic.js

```diff
@@ -36,7 +69,17 @@ const newArray = function newArray(length, defaultValue) {
 // The function you write should return
 // the object passed in as an argument or the object created
 const addMethod = function addMethod(propertyName, method, obj) {
+  let newObj = {};
+  newObj = obj;
+
+  if (obj.propertyName !== null) {
+    newOnj.propertyName = obj.propertyName;
```

Comment:
This will try to change a property called 'propertyName' rather than the property name that gets passed in as an argument.